awx/main/models/jobs.py

```diff
@@ -268,6 +268,13 @@ class Meta:
             'admin_role',
         ],
     )
+    prevent_instance_group_fallback = models.BooleanField(
+        default=False,
+        help_text=(
+            "If enabled, the job template will prevent adding any inventory or organization "
+            "instance groups to the list of preferred instances groups to run on."
```

Comment:
Isn't the value of `JT.prevent_instance_group_fallback = False` inconsistent with `ask_instance_groups_on_launch`? Perhaps this could be enforced with validation logic...

I guess you could say someone would _want_ the behavior that the job uses org / inventory IGs but only uses the provided ones when launching. But it is a big stretch to imagine that the user would _understand_ the behavior. I don't want to underestimate how big of a documentation and messaging challenging this would be.

Reply:
`JT.prevent_instance_group_fallback` drops you out of the loop in preferred_instance_groups.
`ask_instance_groups_on_launch` (if passed in) [actually prevents us from running perferred_instance_groups](https://github.com/ansible/awx/blob/feature-prompt-on-launch-on-templates/awx/main/models/unified_jobs.py#L385). 

So a non-blank ask_instance_groups_on_launch actually supersedes this option all together. 
What if we add a note in the prompt on launch like:
```
Note setting instance groups from a prompt on launch will set the jobs instances groups to that value and not allow for the default fall back behavior from job template to inventory to organization.
```

Reply:
I modified the help text in this PR to note that setting this new option and providing an empty list of instance groups will result in the global instance groups being selected.